### PR TITLE
docs(schema): document `unionWithSourceData` and `rollupLambda`

### PR DESCRIPTION
**Pull request by hassankhan**
**Check List**
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [x] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required

### CODE REVIEW

File: docs/content/Schema/Reference/pre-aggregations.mdx [hunk 1]
```diff
@@ -843,6 +843,28 @@ cube(`Orders`, {
 });
 ```
 
+### <--{"id" : "Parameters"}--> unionWithSourceData
+
+Pre-aggregations can be configured to behave in real-time fashion by joining an
```
Comment (rpaik) [inline docs/content/Schema/Reference/pre-aggregations.mdx:848]:
```suggestion
Pre-aggregations can be configured to behave in a real-time fashion by joining an
```

File: docs/content/Schema/Reference/pre-aggregations.mdx [hunk 2]
```diff
@@ -911,15 +933,18 @@ columns and pre-aggregated measures in the pre-aggregation definition.
 [Queries][ref-api-ref-query-ref] that can target aggregating indexes must have
 the following characteristics:
 
-1. They cannot make use of any `filters` other than for those dimensions that included in that index
+1. They cannot make use of any `filters` other than for those dimensions that
```
Comment (rpaik) [inline docs/content/Schema/Reference/pre-aggregations.mdx:936]:
```suggestion
1. They cannot make use of any `filters` other than for those dimensions that are
```

File: docs/content/Schema/Reference/pre-aggregations.mdx [hunk 3]
```diff
@@ -911,15 +933,18 @@ columns and pre-aggregated measures in the pre-aggregation definition.
...
 
 Queries that do not have the above can still make use of regular indexes so
 performance can still be optimized.
 
 Basically, the aggregating index is a rollup of data in a rollup table itself.
-Data needs to be downloaded from the source DB as many times as many rollups you have.
-Given that having a single rollup and multiple aggregate indexes gives you pretty same performance from the Cube Store side but multiple times less cost from a data warehouse perspective.
+Data needs to be downloaded from the source DB as many times as many rollups you
+have. Given that having a single rollup and multiple aggregate indexes gives you
+pretty same performance from the Cube Store side but multiple times less cost
```
Comment (rpaik) [inline docs/content/Schema/Reference/pre-aggregations.mdx:946]:
```suggestion
pretty much the same performance from the Cube Store side but multiple times less cost
```

**Comment by keydunov** [comment]
@hassankhan I've merged https://github.com/cube-js/cube.js/pull/5321 feel free to link to it.

File: docs/content/Schema/Reference/pre-aggregations.mdx [hunk 4]
```diff
@@ -301,6 +302,16 @@ cube('Orders', {
 
 </InfoBox>
 
+<h4 id="parameters-type-rolluplambda">rollupLambda</h4>
+
+A `rollupLambda` pre-aggregation is a special type of pre-aggregation that can
+combine data from data sources and other rollups. It is extremely useful for
+scenarios where data accuracy is a priority, such as exchange rates for example.
```
Comment (keydunov) [inline docs/content/Schema/Reference/pre-aggregations.mdx:309]: Just generally real-time use cases?

**Pull request merged**